modules/prebid/optimization/config/rules-engine-schema.json

```diff
@@ -0,0 +1,131 @@
+          },
+          "version": {
+            "type": "string"
+          },
+          "modelGroups": {
+            "type": "array",
+            "minItems": 1,
+            "items": {
+              "type": "object",
+              "properties": {
+                "weight": {
+                  "type": "number",
+                  "minimum": 0,
+                  "maximum": 100
+                },
+                "analyticsKey": {
+                  "type": "string"
+                },
+                "version": {
+                  "type": "string"
+                },
+                "schema": {
+                  "type": "array",
+                  "minItems": 1,
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "function": {
+                        "type": "string",
+                          "enum": ["deviceCountry", "dataCenters", "channel", "eidAvailable", "userFpdAvailable", "fpdAvail", "gppSid", "tcfInScope", "percent", "prebidKey", "domain", "bundle", "deviceType"]
+                      },
+                      "args": {
+                        "type": "array"
+                      }
+                    },
+                    "required": ["function"]
+                  }
+                },
+                "default": {
+                  "description": "Just like rules[i].results but optional",
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "function": {
+                        "type": "string",
+                        "enum": ["excludeBidders", "includeBidders", "logATag"]
+                      },
+                      "args": {
+                        "type": "array"
+                      }
+                    },
+                    "required": ["function"]
+                  }
+                },
+                "rules": {
+                  "type": "array",
+                  "minItems": 1,
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "conditions": {
+                        "type": "array",
+                        "minItems": 1,
+                        "items": {
+                          "type": "string"
+                        }
+                      },
+                      "results": {
+                        "type": "array",
+                        "minItems": 1,
+                        "items": {
+                          "type": "object",
+                          "properties": {
+                            "function": {
+                              "type": "string",
+                              "enum": ["excludeBidders", "includeBidders", "logATag"]
+                            },
+                            "args": {
+                              "type": "array"
+                            }
```

Comment:
I see in the spec that result function args are not always an array like schema functions. I will talk to the committee to see if we can change that. Let's leave what you have here.